### PR TITLE
New toplevel structure and package name: climatevision

### PR DESCRIPTION
**Pull request by bgrundmann**
Rename the package to `climatevision`.  Move the so far existing core calculation into a sub package called generator.  So what used to be `generatorcore` is now in `climatevision.generator`.  

This is done so we have a separate place to put support code that will be shared between knud and the devtools. (E.g. `climatevision.serverpcs` or similar).

Furthermore I put the `src` of the package (as opposed to `devtool` and the like) into a separate directory called `src`.  Why?  Mainly because of the bullet point on `import parity` in this article https://blog.ionelmc.ro/2014/05/25/python-packaging/.

Doing the rename I found a bunch of places where we did e.g. `import <somemodule>` instead of `from . import <somemodule>` *INSIDE* of what used to be `generatorcore`.  This is *BAD*!  Why? Because it means that if you run based on the code in the current directory instead of the installed package, you would end up partially running the code in the current directory and partially the installed code.

With all the changes devtool will always only used the installed package!  That should hopefully make all of this easier to understand BUT will require that after making changes inside the `src` folder you have to run `poetry install` (I might make that part of `ready to rock`, to make sure that nobody forgets).

## Update 

This is a little tricky.  I recommend you get rid of your old virtualenv and create a fresh one.  That way you don't have the old generatorcore install laying around anymore.  I did that like this (all in the root of the checkout):

```
poetry env info
poetry env remove --all
python3 -m venv .venv
poetry env use .venv/bin/python
poetry install    
poetry shell
```

## Ready to rock

```
(.venv) --- localzero/localzero-generator-core ‹new-toplevel-structure› » python devtool.py ready_to_rock                         
No configuration file found.
pyproject.toml file found at /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core.
Loading pyproject.toml file at /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core/pyproject.toml
Assuming Python platform Darwin
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
stubPath /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core/typings is not a valid directory.
Searching for source files
Found 137 source files
pyright 1.1.270
0 errors, 0 warnings, 0 informations 
Completed in 2.181sec
=========================================================================== test session starts ============================================================================
platform darwin -- Python 3.10.6, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core
plugins: cov-3.0.0
collected 31 items                                                                                                                                                         

tests/test_devtool_commands.py ...............                                                                                                                       [ 48%]
tests/test_end_to_end.py ...........                                                                                                                                 [ 83%]
tests/test_entries.py .                                                                                                                                              [ 87%]
tests/test_refdata.py ....                                                                                                                                           [100%]

============================================================================ 31 passed in 4.37s ============================================================================
Trim Trailing Whitespace.................................................Passed
Mixed line ending........................................................Passed
Check for case conflicts.................................................Passed
Check Yaml...............................................................Passed
Check for added large files..............................................Passed
Don't commit to branch...................................................Passed
black....................................................................Passed
You are ready to rock and save the climate at c1480bc80556d0175705652061e3d14620dc72d0, but don't forget to copy paste the above into your pull request
```

### CODE REVIEW

File: src/lzcv/generator/agri2018/dataclasses.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
+# pyright: strict
```
Comment (Jeniffere) [inline src/lzcv/generator/agri2018/dataclasses.py:1]: I think, this is an old file, which you accidently commited

File: src/lzcv/generator/agri2030/dataclasses.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
+# pyright: strict
```
Comment (Jeniffere) [inline src/lzcv/generator/agri2030/dataclasses.py:1]: I think, this is an old file, which you accidently commited

File: src/lzcv/generator/agri2030/energy_demand.py [hunk 3]
```diff
@@ -0,0 +1,977 @@
+# pyright: strict
```
Comment (Jeniffere) [inline src/lzcv/generator/agri2030/energy_demand.py:1]: I think, this is an old file, which you accidently commited

File: src/lzcv/generator/agri2030/energy_source.py [hunk 4]
```diff
@@ -0,0 +1,406 @@
+# pyright: strict
```
Comment (Jeniffere) [inline src/lzcv/generator/agri2030/energy_source.py:1]: I think, this is an old file, which you accidently commited

**Comment by bgrundmann** [comment]
I had a major merge conflict so entirely plausible — I’ll have a look.
Thanks!

On September 11, 2022, GitHub ***@***.***> wrote:
> @Jeniffere commented on this pull request.
>

> In src/lzcv/generator/agri2030/energy_source.py
> <https://github.com/GermanZero-de/localzero-generator-
> core/pull/278#discussion_r967871915>:
>

> @@ -0,0 +1,406 @@ +# pyright: strict
> I think, this is an old file, which you accidently commited
>

—
Reply to this email directly, view it on GitHub
<https://github.com/GermanZero-de/localzero-generator-
core/pull/278#pullrequestreview-1103341331>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/AAHQUJIVNHGLC2CK67K5KK3V5YSPNANCNFSM6AAAAAAQJ4FB7Q>.
You are receiving this because you authored the thread.Message ID:
<GermanZero-de/localzero-generator-
***@***.***>

File: src/lzcv/generator/agri2030/energy_source.py (hunk 4 above)
Comment (bgrundmann) [reply src/lzcv/generator/agri2030/energy_source.py]: Same was true for energy_demand.py in the same file.  Thanks again.

**Pull request merged**